### DISCUSSION TITLE
Important integrations for all users.

### DISCUSSION THREAD

**Discussion by sdomeradzki** [breezefront/community#9] [3. Integrations]
Hi,
Below I listed most popular and most useful extension:

- Improved Configurable Product from Firebearstudio
- Advanced Custom Options from Mageworx
- Mirasvit Blog Extensions

Two first extensions are really helpful for all.

**Comment by dimdimich** [comment]
Thank you for your suggestion. 

Recently we had a discussion with Firebearstudio about adding support to Breeze frontend. We will have more information next week on that issue.

As for Advanced Custom Options from Mageworx we already working on integration. Most likely we will be able to publish the initial integration next week. 

We will check the blog module but if you are not forced to use Mirasvit Blog module you can consider using Magefun blog. We already added integration for it.

**Comment by sdomeradzki** [comment]
Great news. I will be wait for this :)

**Comment by sdomeradzki** [comment]
Hi all,
Did you have maybe any information about integrations? Any progres?? :) @dimdimich @vovayatsyuk

**Comment by dimdimich** [comment]
Sorry for delay. We still waiting for update from Firebear. Please contact them from your side also. Probably that will help a bit to get their final decision faster.

As for Custom Options from Mageworx we still waiting for our developers.

**Reply by risecommerce** [reply]
Hello @dimdimich 
I saw mageworx/advancedproductoptions support on breezefront extensions page and I have a question.
I want to ask if you guys are still working on something because I am facing a JS issue while using advancedproductoptions dependency options. 
It do not allow to show dependent options while selecting it works great in the luma theme also works well on argento luma theme but not working in breeze evolution theme.
Can you help me with this?

Thanks

**Comment by fvlasie** [comment]
I am also seeing the same issue as @risecommerce with the Mageworx Advanced Product Options module.

**Comment by fvlasie** [comment]
Here is the Javascript error I get in browser console when clicking an option:

```
[Error] TypeError: undefined is not an object (evaluating 'name.indexOf')

	findOptionId (price-utils.js:98)
	processSwatchLabel (swatches.js:104)
	update (swatches.js:159)
	(anonymous function) (base.js:406)
	(anonymous function)
	d (cash-8.1.4.min.js:1:1941)
	onOptionChanged (base.js:403)
	onOptionChanged
	et (cash-8.1.4.min.js:1:16175)
	dispatchEvent
	(anonymous function) (cash-8.1.4.min.js:1:14411)
	d (cash-8.1.4.min.js:1:2014)
	(anonymous function) (features.js:881)
	d (cash-8.1.4.min.js:1:2014)
	_addValueDescription (features.js:795)
	firstRun (features.js:128)
	runUpdater (base.js:189)
	addUpdater (base.js:127)
	(anonymous function) (3609:10)
	(anonymous function) (define.js:64)
	Global Code (3609:2)
	(anonymous function) (turbolinks-5.2.0.min.js:5:19765)
	(anonymous function) (turbolinks-5.2.0.min.js:5:17913)
	(anonymous function) (turbolinks-5.2.0.min.js:5:17483)
	(anonymous function) (turbolinks-5.2.0.min.js:5:16340)
	(anonymous function) (turbolinks-5.2.0.min.js:5:17432)
	(anonymous function) (turbolinks-5.2.0.min.js:5:16247)
	(anonymous function) (turbolinks-5.2.0.min.js:5:22774)
	(anonymous function) (turbolinks-5.2.0.min.js:5:22500)
	(anonymous function) (turbolinks-5.2.0.min.js:6:762)
	(anonymous function) (turbolinks-5.2.0.min.js:5:27164)
	(anonymous function) (turbolinks-5.2.0.min.js:5:29808)
```

**Comment by fvlasie** [comment]
I have submitted an issue for this here: https://github.com/breezefront/module-breeze-mageworx-advancedproductoptions/issues/1

**Comment by fvlasie** [comment]
swissuplabs has fixed the Mageworx Advanced Product Options module here: https://github.com/breezefront/module-breeze-mageworx-advancedproductoptions

**Comment by risecommerce** [comment]
Hi @dimdimich ,
I am getting this error on clicking mageworx-swatch-option in MageWorx_OptionFeatures/js/swatches/additional.js
Using breeze blank theme 
Can you help me with this ?

```
product2.js:1599 Uncaught TypeError: Cannot read properties of undefined (reading 'removeCandidate')
    at Class._removeCandidateForReplacement (product2.js:1599:26)
    at Class.elementChange (product2.js:1386:30)
    at Class.update (product2.js:1317:22)
    at Class.<anonymous> (product1.js:406:29)
    at Function.d [as each] (core1.js:1:1936)
    at Class.onOptionChanged [as optionChanged] (product1.js:403:15)
    at HTMLSelectElement.et (core1.js:1:16170)
    at HTMLSelectElement.<anonymous> (core1.js:1:14397)
    at d (core1.js:1:2009)
    at e.each (core1.js:1:2085)
```
Thanks

**Reply by tangar76** [reply]
Can you please tell us how we have to configure Mageworx module to get this JS error? Or maybe you can share with us you store url and pint to the page where mentioned JS error occurs.

**Reply by risecommerce** [reply]
Hi @tangar76,
Thanks for your response
Here is the screen shot where I explained the steps

[https://tinyurl.com/2zrmhblu](https://tinyurl.com/2zrmhblu)
[https://www.awesomescreenshot.com/video/17884230?key=f48f5c0f6dd430d6efe6ebfed090534a](https://www.awesomescreenshot.com/video/17884230?key=f48f5c0f6dd430d6efe6ebfed090534a)
Getting error when click on Shade card swatches .
With luma it is working fine 
here is url 
product url :https://support.pushplinen.com/beddings/bedding-sets/sheet-sets/stripe-sheet-set-triple-embroidery-border

**Reply by tangar76** [reply]
@risecommerce your screenshot and video were very helpful. Thank you.

Get master branchs of [breezefront/module-breeze](https://github.com/breezefront/module-breeze) and [breezefront/module-breeze-mageworx-advancedproductoptions](https://github.com/breezefront/module-breeze-mageworx-advancedproductoptions). We committed some changed there regarding your issue.

Please check check if it solved the error.

**Reply by risecommerce** [reply]
> @risecommerce your screenshot and video were very helpful. Thank you.
> 
> Get master branchs of [breezefront/module-breeze](https://github.com/breezefront/module-breeze) and [breezefront/module-breeze-mageworx-advancedproductoptions](https://github.com/breezefront/module-breeze-mageworx-advancedproductoptions). We committed some changed there regarding your issue.
> 
> Please check check if it solved the error.

Hi @tangar76,
Thanks for your response,

after applying these changes I can see following errors in imageReplacer-register.js 
`Uncaught TypeError: Cannot read properties of undefined (reading 'addCandidate')
    at product3.js:178:32
    at Function.lr [as findKey] (core.js:13:8515)
    at product3.js:177:19
    at window.require (core1.js:532:21)
    at window.define (core1.js:536:48)
    at product3.js:171:1` 
I am using 
"swissup/module-breeze": "2.5.5",
"swissup/module-breeze-mageworx-advancedproductoptions": "1.1.2"
and magento version : 2.4.5-p1
If you want I can share store credentials  too.
Thanks

**Reply by tangar76** [reply]
Please apply changes from this commit - [ff4f25d3](https://github.com/breezefront/module-breeze-mageworx-advancedproductoptions/commit/ff4f25d39bc114e3a4444a7cb8c0054a57eea957)

**Reply by risecommerce** [reply]
> Please apply changes from this commit - [ff4f25d3](https://github.com/breezefront/module-breeze-mageworx-advancedproductoptions/commit/ff4f25d39bc114e3a4444a7cb8c0054a57eea957)
After applying it
`product2.js:1626 Uncaught TypeError: Cannot read properties of undefined (reading 'removeCandidate')
    at Class._removeUnselectedCandidates (product2.js:1626:38)
    at Class.elementChange (product2.js:1392:26)
    at Class.update (product2.js:1317:22)
    at Class.<anonymous> (product1.js:406:29)
    at Function.d [as each] (core1.js:1:1936)
    at Class.onOptionChanged [as optionChanged] (product1.js:403:15)
    at HTMLSelectElement.et (core1.js:1:16170)
    at HTMLSelectElement.<anonymous> (core1.js:1:14397)
    at d (core1.js:1:2009)
    at e.each (core1.js:1:2085)`

**Reply by tangar76** [reply]
Please disable JS merge for store https://support.pushplinen.com/ and check order of loaded JS files. It has to be exactly as it is in [breeze_default__mageworx_optionfeatures.xml](https://github.com/breezefront/module-breeze-mageworx-advancedproductoptions/blob/master/view/frontend/layout/breeze_default__mageworx_optionfeatures.xml).

**Reply by risecommerce** [reply]
> is

Thanks @tangar76 ,
It is working now but on click shade card it keeps reloading and getting error 
`imageReplacer.js:93 Uncaught (in promise) TypeError: result.first is not a function
    at imageReplacer.js:93:28`

**Reply by tangar76** [reply]
It looks like your module Breeze doesn't have latest commits.

**Reply by tangar76** [reply]
Please, download master branch of module Breeze and update .

**Reply by risecommerce** [reply]
Hello @tangar76 ,
Thanks for your reply my module breeze is upto date (I did it again) but it still did not worked.
Getting same error (imageReplacer.js:93 Uncaught (in promise) TypeError: result.first is not a function at imageReplacer.js:93:28)
due to this Add to cart function is not working in iPhone Safari Browser.
Can you furthur help me with it ?

**Reply by risecommerce** [reply]
hi @tangar76 
@dimdimich 
I have updated modules with latest commits as you suggested
but getting  following errors on Product page when click on Shade Card
`Uncaught DOMException: Failed to execute 'querySelectorAll' on 'Document': '#select_2415 option:selected' is not a valid selector.
` 
and nothing works on cart product edit page with error 
`Uncaught TypeError: Updater is not a constructor
    at configurable-customer-data.js:25:15
    at window.require (define.js:73:30)
    at (index):156:62
    at Array.map (<anonymous>)
    at (index):156:48`
Can you help me with that?
Thanks

**Reply by tangar76** [reply]
Unfortunately there is no way to solve this error without applying patch to MageWorx file.
Please apply this change https://github.com/breezefront/module-breeze-mageworx-advancedproductoptions#required-patches

It would be nice if you could contact MageWorx team from you side (since you are there customer) and ask to include change from that patch in to newest version of their extension.

**Reply by risecommerce** [reply]
@tangar76 ,
After applying patch it works on product page that is fine .

Now the problem is  
when customer want to edit product after adding into cart  the options are not working with the  following error..

`configurable-customer-data.js:25 Uncaught TypeError: Updater is not a constructor
    at configurable-customer-data.js:25:15
    at window.require (define.js:73:30)
    at (index):158:62
    at Array.map (<anonymous>)
    at (index):158:48`
Mageworks is working fine with Luma and Magento Blank theme .

**Reply by tangar76** [reply]
It looks like there is a missing breeze integration for edit configurable product item in cart. Our team will work on it.

**Reply by vovayatsyuk** [reply]
@risecommerce, I've just tagged new breeze releases (module-breeze and blank theme). Please update and check if it will work fine for you.

**Comment by fvlasie** [comment]
Mageworx has agreed to add the patches to the next release of Advanced Product Options!
